braindecode/models/tidnet.py

```diff
@@ -0,0 +1,251 @@
+        x = self.spatial(x)
+        return self.extract_features(x)
+
+
+class TIDNet(nn.Module):
+    """Thinker Invariance DenseNet model from [TIDNet]_
+
+    Parameters
+    ----------
+    n_classes : int
+        Number of classes.
+    in_chans : int
+        Number of EEG channels.
+    input_window_samples : int
+        Number of samples.
+    s_growth : int
+        DenseNet-style growth factor (added filters per DenseFilter)
+    t_filters : int
+        Number of temporal filters.
+    drop_prob : float
+        Dropout probability
+    pooling : int
+        Max temporal pooling (width and stride)
+    temp_layers : int
+        Number of temporal layers
+    spat_layers : int
+        Number of DenseFilters
+    temp_span : float
+        Percentage of input_window_samples that defines the temporal filter length:
+        temp_len = ceil(temp_span * input_window_samples)
+        e.g A value of 0.05 for temp_span with 1500 input_window_samples will yield a temporal
+        filter of length 75.
+    bottleneck : int
+        Bottleneck factor within Densefilter
+    summary : int
+        Output size of AdaptiveAvgPool1D layer. If set to -1, value will be calculated
+        automatically (input_window_samples // pooling).
+
+    Notes
+    -----
+    Code adapted from: https://github.com/SPOClab-ca/ThinkerInvariance/blob/master/
+
+    References
+    ----------
+    .. [TIDNet] Kostas, D. & Rudzicz, F.
+        Thinker invariance: enabling deep neural networks for BCI across more people.
+        J. Neural Eng. 17, 056008 (2020).
+        doi: 10.1088/1741-2552/abb7a7.
+    """
+    def __init__(self, in_chans, n_classes, input_window_samples, s_growth=24, t_filters=32,
+                 drop_prob=0.4, pooling=15, temp_layers=2, spat_layers=2, temp_span=0.05,
+                 bottleneck=3, summary=-1):
+        super().__init__()
+        self.n_classes = n_classes
+        self.in_chans = in_chans
+        self.input_window_samples = input_window_samples
+        self.temp_len = ceil(temp_span * input_window_samples)
+
+        self.dscnn = _tidnet_features(s_growth=s_growth, t_filters=t_filters, in_chans=in_chans,
+                                      input_window_samples=input_window_samples,
+                                      drop_prob=drop_prob, pooling=pooling, temp_layers=temp_layers,
+                                      spat_layers=spat_layers, temp_span=temp_span,
+                                      bottleneck=bottleneck, summary=summary)
+
+        self._num_features = self.dscnn.num_features
+
+        self.classify = self._create_classifier(self.num_features, n_classes)
+
+    def _create_classifier(self, incoming, n_classes):
+        classifier = nn.Linear(incoming, n_classes)
+        init.xavier_normal_(classifier.weight)
+        classifier.bias.data.zero_()
+        return nn.Sequential(nn.Flatten(start_dim=1), classifier, nn.LogSoftmax(dim=-1))
+
+    def forward(self, x, **kwargs):
```

Comment:
docstring of public method missing

Reply:
do we really want docstring for forward methods? E.g., PyTorch doesn't do this: 
https://github.com/pytorch/pytorch/blob/bac4cfd54d44aa0fbc574e6561b878cb406762cc/torch/nn/modules/conv.py#L442

Neither does torchvision:
https://github.com/pytorch/vision/blob/d391a0e992a35d7fb01e11110e2ccf8e445ad8a0/torchvision/models/densenet.py#L215

Reply:
but I don't think we need **kwargs here? this could be removed?

Reply:
I would say here we could document the shapes of x for this network module? but I will not fight hard for it

Reply:
ok since you're not gonna fight for it and I think we should always have same accepted input shapes: #examples x #EEGchannels x #time (and optionally except empty last dim), we can just keep without docstring but remove **kwargs